src/dialer/index.js

```diff
@@ -118,6 +124,10 @@ class Dialer {
    * @returns {Promise<Connection>}
    */
   async connectToPeer (peer, options = {}) {
+    const { id } = getPeer(peer)
+    if (this.connectionManager && await this.connectionManager.gater.interceptPeerDial(id)) {
```

Comment:
Is there a case when `this.connectionManager` will be undefined? In the upgrader, it seems like it isn't treated that way.